client.go

```diff
@@ -150,18 +158,24 @@ func (c *detaClient) request(i *requestInput) (*requestOutput, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer res.Body.Close()
-
-	b, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, err
-	}
 
+	// request output
 	o := &requestOutput{
 		Status: res.StatusCode,
 		Header: res.Header,
 	}
 
+	if i.ShouldReadBody && res.StatusCode >= 200 && res.StatusCode <= 299 {
```

Comment:
shouldn't be called `ShouldReadBody` if you are not reading the body when `ShouldReadBody` is `true`, can be called `SkipReadBody` ?

Reply:
can also be called `ReturnReadCloser` probably better